src/path-optimization/spline-gradient-based.cc

```diff
@@ -460,7 +461,7 @@ PathVectorPtr_t SplineGradientBased<_PB, _SO>::optimize(
     if (!this->validateBounds(splines, boundConstraint).empty())
       throw std::invalid_argument("Input path does not satisfy joint bounds");
   }
-  LinearConstraint boundConstraintReduced(boundConstraint.PK.rows(), 0);
```

Comment:
This change is rather surprising to me. I would expect that the reduced constraint has the same number of rows as the full one.

Is this a typo? If not, can you add a comment?

Reply:
Any comment on this one?

Reply:
Analysing the code, I did not find where boundConstraint.PK was initialized since method decompose is never called for this instance.

Reply:
For a reason that I do not understand, the above comment was tagged as "pending" and therefore not visible by anybody else than me.